solana/gateway-ts/src/lib/util.ts

```diff
@@ -1,81 +1,78 @@
-import {
-  AccountInfo,
-  Commitment,
-  Connection,
-  DataSlice,
-  GetProgramAccountsFilter,
-  PublicKey,
-} from "@solana/web3.js";
+import { Connection, PublicKey } from "@solana/web3.js";
 import {
   GATEKEEPER_NONCE_SEED_STRING,
   GATEWAY_TOKEN_ADDRESS_SEED,
   PROGRAM_ID,
   SOLANA_COMMITMENT,
 } from "./constants";
-import { GatewayToken, ProgramAccountResponse, State } from "../types";
+import { GatewayToken, State } from "../types";
 import { GatewayTokenData, GatewayTokenState } from "./GatewayTokenData";
 import { mapEnumToFeatureName, NetworkFeature } from "./GatewayNetworkData";
-import { encode } from "bs58";
 
 /**
  * Derive the address of the gatekeeper PDA for this gatekeeper
  * @param authority The gatekeeper
  * @param network The network
  */
-export const getGatekeeperAccountAddress = async (
+export const getGatekeeperAccountAddress = (
   authority: PublicKey,
   network: PublicKey
-): Promise<PublicKey> => {
-  const publicKeyNonce = await PublicKey.findProgramAddress(
+): PublicKey =>
+  PublicKey.findProgramAddressSync(
     [
       authority.toBuffer(),
       network.toBuffer(),
       Buffer.from(GATEKEEPER_NONCE_SEED_STRING, "utf8"),
     ],
     PROGRAM_ID
-  );
-  return publicKeyNonce[0];
-};
+  )[0];
 
 /**
  * Derive the address of the gateway token PDA for this owner address and optional seed.
  * @param owner The owner of the gateway token
  * @param gatekeeperNetwork The network of the gateway token
- * @param seed An 8-byte seed array, used to add multiple tokens to the same owner. Must be unique to each token, if present
+ * @param index The index of the gateway token (default 0)
  */
-export const getGatewayTokenAddressForOwnerAndGatekeeperNetwork = async (
+export const getGatewayTokenAddressForOwnerAndGatekeeperNetwork = (
   owner: PublicKey,
   gatekeeperNetwork: PublicKey,
-  seed?: Uint8Array
-): Promise<PublicKey> => {
-  const additionalSeed = seed
-    ? Buffer.from(seed)
-    : Buffer.from([0, 0, 0, 0, 0, 0, 0, 0]);
-  if (additionalSeed.length != 8) {
+  index = 0
+): PublicKey => {
+  // The index is converted to an 8-byte uint array. Ensure no overflow here.
+  if (index > 2 ** (8 * 8)) {
```

Comment:
I think there is no way that `index: number` can ever be more then `2 ** (8 * 8)`. I've even a little bit worried that the right side overflows...

Just tested in Browser. Maybe it's wrong:
```
2 ** (8 * 8)
18446744073709552000
const i = 2 ** (8 * 8)
undefined
i
18446744073709552000
typeof(i)
'number'
Number.MAX_SAFE_INTEGER
9007199254740991
```